api4/webhook.go

```diff
@@ -147,7 +147,14 @@ func getIncomingHooks(c *Context, w http.ResponseWriter, r *http.Request) {
 			return
 		}
 
-		hooks, err = c.App.GetIncomingWebhooksForTeamPage(teamId, c.Params.Page, c.Params.PerPage)
+		var userId string
+
+		// Add userId to only return your own webhooks if there is no permission to manage others.
+		if !c.App.SessionHasPermissionToTeam(c.App.Session, teamId, model.PERMISSION_MANAGE_OTHERS_INCOMING_WEBHOOKS) {
```

Comment:
Let's invert this check and assignment so that it "reads":
```go
userId := c.App.Session.UserId

if hasRequiredPermission {
    // Relax restrictions
    userId = ""
}
```

We've had issues before where the logic was backwards and bugs were later introduced by accident.

Reply:
✅Done